tools/python/upload_and_run_browserstack_tests.py

```diff
@@ -0,0 +1,93 @@
+import argparse
+import json
+import time
+
+import requests
+
+parser = argparse.ArgumentParser("Upload and run BrowserStack tests")
+
+parser.add_argument("--id", type=str, help="BrowserStack user ID")
+parser.add_argument("--token", type=str, help="BrowserStack token")
+parser.add_argument("--test_platform", type=str, help="Testing platform, i.e., XCUITest or Espresso")
+parser.add_argument("--app_apk_path", type=str, help="Path to the app APK")
+parser.add_argument("--test_apk_path", type=str, help="Path to the test suite APK")
+parser.add_argument(
+    "--devices",
+    type=str,
+    nargs="+",
+    help="List of devices to run the tests on. For more info, "
+    "see https://www.browserstack.com/docs/app-automate/espresso/specify-devices",
+)
+
+args = parser.parse_args()
+
+
+def post_response_to_json(response):
+    if len(response) == 0:
+        raise Exception("No response from BrowserStack")
+    try:
+        json_response = json.loads(response)
+        print("JSON response: ", json.dumps(json_response, indent=2))
+        if "error" in json_response:
+            raise Exception(json_response["error"])
```

Comment:
Are there specific json properties we can check for 'error' instead of matching across the entire response?

Reply:
I switched to just using "response.raise_for_status()" which should raise an error if the response code is not good -- as far as I can tell in the documentation for the REST API responses, there wouldn't be other JSON properties that we should check for